src/osd/OSD.h

```diff
@@ -887,6 +888,10 @@ class OSDService {
   Mutex backfill_request_lock;
   SafeTimer backfill_request_timer;
 
+  // -- Recovery Request Scheduling --
+  Mutex recovery_request_lock;
+  SafeTimer recovery_request_timer;
+
```

Comment:
We should be able to reusing the existing timer and lock (maybe rename it recovery_* instead of backfill_*)